netbout-web/src/main/java/com/netbout/dynamo/DyMessages.java

```diff
@@ -193,23 +195,27 @@ public void post(final String text) throws IOException {
 
     @Override
     public long unread() throws IOException {
-        final Item item = this.region.table(DyFriends.TBL)
-            .frame()
-            .through(
-                new QueryValve()
-                    .withLimit(1)
-                    .withAttributesToGet(DyFriends.ATTR_UNREAD)
-            )
-            .where(DyFriends.HASH, Conditions.equalTo(this.bout))
-            .where(DyFriends.RANGE, Conditions.equalTo(this.self))
-            .iterator().next();
-        final long unread;
-        if (item.has(DyFriends.ATTR_UNREAD)) {
-            unread = Long.parseLong(item.get(DyFriends.ATTR_UNREAD).getN());
-        } else {
-            unread = 0L;
+        try {
+            final Item item = this.region.table(DyFriends.TBL)
+                .frame()
+                .through(
+                    new QueryValve()
+                        .withLimit(1)
+                        .withAttributesToGet(DyFriends.ATTR_UNREAD)
+                )
+                .where(DyFriends.HASH, Conditions.equalTo(this.bout))
+                .where(DyFriends.RANGE, Conditions.equalTo(this.self))
+                .iterator().next();
```

Comment:
@original-brownbear i agree completely, but, for me, it seems like a !hasNext() is not expected here, that why i catched this. May be i didn't catch some logic here, no problems, it will be fixed.